### DISCUSSION TITLE
How to implement Storage Commitment to ensure for files are stored on SCP server

### DISCUSSION THREAD

**Discussion by ashishsinghintero** [fo-dicom/fo-dicom#1234] [Q&A]
Hi Everyone,

   I have a case where I have send about 15 files which are part of same series to SCP Server. Once I got the success response after C-Store I have deleted those files from my local client. When I checked SCP server due to storage space full, all 15 files are not stored in SCP server. So I want to implement Storage commitment in my application to ensure data is stored properly in SCP server.
 So if it is possible can someone help me with some code example how to implement Storage Commitment and verify all files are stored properly in SCP server.

Thanks in advance for your help.

**Comment by mrbean-bremen** [comment]
See also #955, specifically the link there.

**Comment by gofal** [comment]
Storage commitment is hard to implement within a library. It's like implementing a query server in fo-dicom. The library could give some implementations of the message, but the main thing has to be implemented within the application logic.
Because The SCP gets a StoreageCommitment Request containg some InstanceUIDs ( a simple n-action request) . Then the server has to verfy that the files are actually present (cannot be implemented in fo-dicom). The result dataset is then a simple list if the instances together with the status (present or missing). The scp has to send the n-event-report to the client. but this is often handled on a separate association, so the SCP needs some configuration environment, on which ip and port the client is listening (cannot be implemented in fo-dicom).

You see, Storage Commitment is rather a workflow and part of an application, not a message or simple event that would be part of a library.

Once, a fo-dicom-server will be implemented, storage commitment could be part of that.
